MATHPlugin.glyphsPlugin/Contents/Resources/plugin.py

```diff
@@ -658,6 +670,57 @@ def draw_(self, layer, options):
         except:
             _message(f"Drawing anchors failed:\n{traceback.format_exc()}")
 
+        draw_variants = self.defaults[f"{PLUGIN_ID}.toggleShowGV:"]
+        if not draw_variants:
+            return
+
+        plugin_data = layer.parent.userData[f"{PLUGIN_ID}.variants"]
+        if not plugin_data:
+            return
+
+        scale = 1 / options["Scale"]
+        if va := plugin_data.get(V_ASSEMBLY_ID):
+            self._draw_v_assembly(va, layer, scale)
+        
+        if ha := plugin_data.get(H_ASSEMBLY_ID):
+            self._draw_h_assembly(ha, layer, scale)
+    
+    @objc.python_method
+    def _draw_h_assembly(self, recipe, layer, width):
+        save()
+        translate(layer.width, layer.bounds.origin.y)
+        for gref, flag, bot, top in recipe:
+            if isinstance(gref, GSGlyphReference):
+                glyph = gref.glyph
+            else:
+                glyph = layer.parent.parent.glyphs[gref]
+            gref_layer = glyph.layers[layer.layerId]
+            translate(-gref_layer.bounds.origin.x, 0)
+            NSColor.blueColor().set()
+            path = gref_layer.completeBezierPath
+            path.setLineWidth_(width)
+            path.stroke()
+            translate(gref_layer.bounds.size.width + gref_layer.bounds.origin.x, 0)
```

Comment:
Shouldn’t this be using the layer width instead of the bounding box? Assembly layer have -ve side bearings to be used for `startConnectorLength`/`endConnectorLength` in the math table (to allow for overlap), but this code will draw them without overlap. You probably should use `_getMetrics()` as it has fallback for unset `layer.vertWidth`).

Reply:
Actually, `bot` and `top` above are the start and end connectors, so you should use that together with the layer width/vertWidth.

Reply:
Actually (sorry), you should ignore these, since this is the maximum overlap. You need to use `MinConnectorOverlap` from the MATH constants.